src/modules/pedidos/components/Timeline.vue

```diff
@@ -0,0 +1,94 @@
+<script setup>
+import { onMounted, reactive } from "vue";
+
+const status = reactive({
+  pagamento: false,
+  preparo: false,
+  saiu: false,
+  entregue: false,
+  done: false,
+});
+
+function handleStatus() {
+  setTimeout(() => {
+    status.pagamento = true;
+  }, 1000);
+
+  setTimeout(() => {
+    status.preparo = true;
+  }, 2000);
+
+  setTimeout(() => {
+    status.saiu = true;
+  }, 3000);
+
+  setTimeout(() => {
+    status.entregue = true;
+    status.done = true;
+  }, 4000);
+}
+
+onMounted(() => {
+  handleStatus();
+});
+</script>
+
+<template>
+  <v-sheet align="center" class="elevation-0">
+    <v-progress-linear
+      color="#fe8b05"
+      style="
+        position: relative;
+        left: -170px;
+        width: 150%;
+        transform: rotate(90deg);
+      "
+      v-if="!status.done"
+      indeterminate
+    ></v-progress-linear>
+    <v-timeline line-color="#fe8b05" side="end">
+      <v-timeline-item
+        :dot-color="status.pagamento ? 'green-darken-4' : 'grey'"
+      >
+        <span
+          :class="[
+            'font-weight-bold',
+            status.pagamento ? 'text-green-darken-4' : 'text-grey',
+          ]"
+        >
+          Pagamento recebido
+        </span>
+      </v-timeline-item>
+      <v-timeline-item :dot-color="status.preparo ? 'green-darken-4' : 'grey'">
+        <span
+          :class="[
+            'font-weight-bold',
+            status.preparo ? 'text-green-darken-4' : 'text-grey',
```

Comment:
vocês poderiam criar uma função que, retorne essas classes baseadas no valor passado, aí poderiam reutilizar em todos os passos

```
function getStatusClass(value) {
  return value ? 'text-green-darken-4' : 'text-grey',
}
```

```suggestion
            getStatusClass(status.preparo)
```

Reply:
Funções criadas